hamp_radar/writezarr.py

```diff
@@ -0,0 +1,83 @@
+import xarray as xr
+from pathlib import Path
+from typing import Optional
+
+
+def ideal_zarrchunks(ds: xr.Dataset):
+    """returns ideal size of chunks for variables with these dimensions
+    so that chuunks are about 1MB. Addiontally cocx dimension (size=2)
+    is split into 2 seperate chunks and range dimension (size=512)
+    is sub-divided into 16 smaller chunks (size=32)"""
+    import warnings
+
+    frame = ds.sizes["frame"]
+    if frame > 2621440:
+        warnings.warn(
+            "Warning: number of frames is very large, consider changing ideal chunkshapes",
+            UserWarning,
+        )
+    return {
+        (): (),
+        ("frame",): (frame,),
+        ("frame", "cocx"): (frame, 1),
+        ("frame", "range", "cocx"): (8192, 32, 1),
+        ("frame", "range", "cocx", "fft", "iq"): (512, 32, 1, 256, 2),
+    }
+
+
+def zarr_chunks_encoding(ds: xr.Dataset):
+    ideal_chunks = ideal_zarrchunks(ds)
+    encoding = {
+        x: {"chunks": ideal_chunks[ds[x].dims]}
+        for x in list(ds.coords) + list(ds.data_vars)
+        if ds[x].chunks
+    }
+    return encoding
+
+
+def merge_encodings(e1: dict, e2: dict):
+    for x in set(e1.keys()) & set(e2.keys()):
+        e1[x].update(e2[x])
+    for x in set(e2.keys()) - set(e1.keys()):
+        e1[x] = e2[x]
+    return e1
```

Comment:
This function modifies `e1`, which is potentially dangerous and should be avoided.

Reply:
Better to create a new dictionary? Or we could raise an error instead if any existing encodings exist instead of calling `merge_encodings`?

Reply:
Having a function to merge encodings might be fine (although at the moment I don't see all the implications it might have in my head). But anyways, this function shouldn't modify its inputs but rather return a new dictionary. I'd probably go with a construction similar to:

```python
return {**e1, **e2}
```

This might however need some adaptation, depending on how nested keys should be merged.

Reply:
I think this way deals with the merging appropriately: https://github.com/halo-narval/hamp_radar/pull/18/commits/73c68742f7f839ab2ea785a349d55da01f82bd3b
it creates a new dictionary with keys for each variable from both e1 and e2, but it would throw an error if e1 and e2 had matching keys, e.g. if you tried to set chunks for a variable in both e1 and e2.